### PR TITLE
fix: cosmos sdk account/asset granularity

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

A bunch of fixes WRT account/asset granularity for Cosmos SDK:

- If you are staked on the same opportunity over multiple accounts, it should be displayed for all accounts on the asset page, and clicking it should route you to the modal opened on that specific account, vs. "default account Id" (?) and showing you the opportunities once currently
- Account page should show you the granular balances with no regressions, and most importantly, clicking the opportunity should open the opportunity for the current account, not the highest balance one
- ATOM / OSMO discrimination should work without the need to pass an AccountId down in an asset page

Note, this applies to both ATOM and OSMO assets/staking

tl;dr cleans things up and fixes things

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

Low, though @shapeshift/product might disagree with showing granular opportunities and want to go with the aggregated ones in the asset page

## Testing

- See description

### Asset page

image

### Account page

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3818** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3818" target="_blank">Graphite</a> 
    * **PR #3819** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3819" target="_blank">Graphite</a> 
      * **PR #3820** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3820" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3820?utm_source=stack-comment).

File: src/features/defi/contexts/DefiManagerProvider/DefiCommon.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export type DefiParams = {
 }
 
 export type DefiQueryParams = {
+  accountId?: AccountId
   defaultAccountId?: AccountId
   chainId: ChainId
   highestBalanceAccountAddress?: string
```
Comment (gomesalexandre) [inline src/features/defi/contexts/DefiManagerProvider/DefiCommon.ts:46]: Next up let's remove this, we have a selector for it

**Review by 0xdef1cafe — APPROVED** [review]
